File: manhunt-base/src/main/java/io/github/ytg1234/manhunt/base/compat/ModMenuCompat.java
package io.github.ytg1234.manhunt.base.compat;

import io.github.prospector.modmenu.api.ConfigScreenFactory;
import io.github.prospector.modmenu.api.ModMenuApi;
import io.github.ytg1234.manhunt.base.config.ManhuntConfig;
import me.sargunvohra.mcmods.autoconfig1u.AutoConfig;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;

@Environment(EnvType.CLIENT)
public class ModMenuCompat implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return parent -> AutoConfig.getConfigScreen(ManhuntConfig.class, parent).get();
    }
}

File: manhunt-base/src/main/kotlin/io/github/ytg1234/manhunt/compat/ModMenuCompat.kt
package io.github.ytg1234.manhunt.compat

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import io.github.ytg1234.manhunt.config.ManhuntConfig
import me.shedaniel.autoconfig.AutoConfig
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment

/**
 * Compatibility with ModMenu.
 *
 * @author YTG1234
 */
@Environment(EnvType.CLIENT)
object ModMenuCompat : ModMenuApi {
    /**
     * Used to construct a new config screen instance when your mod's
     * configuration button is selected on the mod menu screen. The
     * screen instance parameter is the active mod menu screen.
     *
     * @return A factory for constructing config screen instances.
     */
    override fun getModConfigScreenFactory() = ConfigScreenFactory {
        AutoConfig.getConfigScreen(ManhuntConfig::class.java, it).get()
    }
}
